forge-gui/res/cardsfolder/a/amber_prison.txt

```diff
@@ -2,9 +2,8 @@ Name:Amber Prison
 ManaCost:4
 Types:Artifact
 K:You may choose not to untap CARDNAME during your untap step.
-A:AB$ Tap | Cost$ 4 T | ValidTgts$ Artifact,Creature,Land | TgtPrompt$ Select target artifact, creature, or land | RememberTapped$ True | AlwaysRemember$ True | SpellDescription$ Tap target artifact, creature, or land. That permanent doesn't untap during its controller's untap step for as long as CARDNAME remains tapped. | StackDescription$ SpellDescription
-R:Event$ Untap | ActiveZones$ Battlefield | ValidCard$ Card.IsRemembered+ActivePlayerCtrl | ActivePhases$ Untap | Layer$ CantHappen
-T:Mode$ Untaps | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ ClearRemembered | Static$ True
-SVar:ClearRemembered:DB$ Cleanup | ClearRemembered$ True
+A:AB$ Tap | Cost$ 4 T | ValidTgts$ Artifact,Creature,Land | TgtPrompt$ Select target artifact, creature, or land | SubAbility$ DBEffect | SpellDescription$ Tap target artifact, creature, or land. That permanent doesn't untap during its controller's untap step for as long as CARDNAME remains tapped. | StackDescription$ SpellDescription
+SVar:DBEffect:DB$ Effect | IsCurse$ True | ReplacementEffects$ DontUntap | Duration$ UntilUntaps | RememberObjects$ Targeted
```

Comment:
@tool4ever you could add `ForgetOnMoved$ Battlefield`, so the effect gets removed when the affected Card is removed